core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -527,31 +529,61 @@ public static TransportType transportType() {
      *
      * <p>This flag is enabled by default for supported platforms. Specify the
      * {@code -Dcom.linecorp.armeria.useOpenSsl=false} JVM option to disable it.
+     *
+     * @deprecated Use {@link #tlsEngineType()} and {@code -Dcom.linecorp.armeria.tlsEngineType=openssl}.
      */
+    @Deprecated
     public static boolean useOpenSsl() {
-        if (useOpenSsl != null) {
-            return useOpenSsl;
+        return tlsEngineType() == TlsEngineType.OPENSSL;
+    }
+
+    /**
+     * Returns the {@link TlsEngineType} that will be used for processing TLS connections.
+     *
+     * <p>The default value of this flag is "openssl", which means the {@link TlsEngineType#OPENSSL} will
+     * be used. Specify the {@code -Dcom.linecorp.armeria.tlsEngineType=<jdk|openssl>} JVM option to override
+     * the default value.
+     */
+    public static TlsEngineType tlsEngineType() {
+        if (tlsEngineType != null) {
+            return tlsEngineType;
         }
         setUseOpenSslAndDumpOpenSslInfo();
-        return useOpenSsl;
+        return tlsEngineType;
     }
 
     private static void setUseOpenSslAndDumpOpenSslInfo() {
-        final boolean useOpenSsl = getValue(FlagsProvider::useOpenSsl, "useOpenSsl");
-        if (!useOpenSsl) {
-            // OpenSSL explicitly disabled
-            Flags.useOpenSsl = false;
-            dumpOpenSslInfo = false;
-            return;
-        }
         if (!OpenSsl.isAvailable()) {
```

Comment:
This seems a regression of https://github.com/line/armeria/pull/2184
Should we check `OpenSsl.isAvailable()` lazily when other constrains are met?

Reply:
Sorry for the late reply 😅 
If the cost of `OpenSsl.isAvailable()` is high, I think it's better to remain as-is. 
What do you think @minwoox, @jrhee17 ?